tcpdump.c

```diff
@@ -2089,7 +2089,7 @@ DIAG_ON_CLANG(assign-enum)
 	if (pcap_setfilter(pd, &fcode) < 0)
 		error("%s", pcap_geterr(pd));
 #ifdef HAVE_CAPSICUM
-	if (RFileName == NULL && VFileName == NULL) {
+	if (RFileName == NULL && VFileName == NULL && pcap_fileno(pd) > 0) {
```

Comment:
That should be `pcap_fileno(pd) != -1`, as `pcap_fileno()` returns -1 on error.  It's *unlikely* that the `pcap_t` file descriptor ends up being 0, and that *shouldn't* happen (programs *should* have the standard input, output, and error open, even if they're all directed to `/dev/null`), but we shouldn't assume it *doesn't* happen.